pkg/scalers/etcd_scaler.go

```diff
@@ -0,0 +1,305 @@
+			meta.cert = config.AuthParams["cert"]
+			meta.key = config.AuthParams["key"]
+			if value, found := config.AuthParams["keyPassword"]; found {
+				meta.keyPassword = value
+			} else {
+				meta.keyPassword = ""
+			}
+			meta.enableTLS = true
+		} else if val != etcdTLSDisable {
+			return fmt.Errorf("err incorrect value for TLS given: %s", val)
+		}
+	}
+
+	return nil
+}
+
+func parseEtcdMetadata(config *ScalerConfig) (*etcdMetadata, error) {
+	meta := &etcdMetadata{}
+	var err error
+	meta.endpoints = strings.Split(config.TriggerMetadata[endpoints], ",")
+	if len(meta.endpoints) == 0 || meta.endpoints[0] == "" {
+		return nil, fmt.Errorf("endpoints required")
+	}
+
+	meta.watchKey = config.TriggerMetadata[watchKey]
+	if len(meta.watchKey) == 0 {
+		return nil, fmt.Errorf("watchKey required")
+	}
+
+	meta.value, err = strconv.ParseFloat(config.TriggerMetadata[value], 64)
+	if err != nil || meta.value <= 0 {
+		return nil, fmt.Errorf("value must be a float greater than 0")
+	}
+
+	meta.activationValue = 0
+	if val, ok := config.TriggerMetadata[activationValue]; ok {
+		meta.activationValue, err = strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("activationValue must be a float")
+		}
+	}
+
+	meta.watchProgressNotifyInterval = 600
+	if val, ok := config.TriggerMetadata[watchProgressNotifyInterval]; ok {
+		meta.watchProgressNotifyInterval, err = strconv.Atoi(val)
+		if err != nil || meta.watchProgressNotifyInterval <= 0 {
+			return nil, fmt.Errorf("watchProgressNotifyInterval must be a int greater than 0")
+		}
+	}
+
+	if err = parseEtcdAuthParams(config, meta); err != nil {
+		return meta, err
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+	return meta, nil
+}
+
+func getEtcdClients(metadata *etcdMetadata) (*clientv3.Client, error) {
+	var tlsConfig *tls.Config
+	var err error
+	if metadata.enableTLS {
+		tlsConfig, err = kedautil.NewTLSConfigWithPassword(metadata.cert, metadata.key, metadata.keyPassword, metadata.ca)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   metadata.endpoints,
+		DialTimeout: 5 * time.Second,
+		TLS:         tlsConfig,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to etcd server: %s", err)
+	}
+
+	return cli, nil
+}
+
+// IsActive determines if we need to scale from zero
+func (s *etcdScaler) IsActive(ctx context.Context) (bool, error) {
+	v, err := s.getMetricValue(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	return v > s.metadata.activationValue, nil
+}
+
+// Close closes the etcd client
+func (s *etcdScaler) Close(context.Context) error {
+	if s.client != nil {
+		return s.client.Close()
+	}
+	return nil
+}
+
+// GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
+func (s *etcdScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
```

Comment:
We've recently refactored this interface, the method signature is:
`func (s *etcdScaler) GetMetrics(ctx context.Context, metricName string) ([]external_metrics.ExternalMetricValue, error)`

Reply:
This change will fix the Static Checks and Validation problems.

Reply:
Thanks for your tip! It's now fixed